docs/shared-ruleset-design.md

```diff
@@ -22,6 +22,16 @@ this easy, or investigate Gazelle). **By default**, we will pull in a proven set
 The above outputs can be released independently, and the versioning/release is something
 the downstream engineers are responsible for.
 
+## Building-blocks
+
+| Canonical name     | Responsibility                                                                                                                                                                                                     | Default behavior                                                                                                                                                       |
+| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Client             | Provides ability to dynamically reference a module federation container and render the exposed route, client-side routing (more client-side feature possible in future)                                            | React with `react-router` to navigate between federated routes, transpiled with SWC and bundled by Webpack.                                                            |
+| Server             | Returns HTML templates for routes that bootstrap the application (features like SSR possible in future). Configurable via CLI args/flags or environment variables.                                                 | Returns HTML page with client bundle and preloaded route federation `remoteEntry`. Internal middleware supported to augment functionality                              |
```

Comment:
What does "internal middleware supported" mean here? Seems like the wrong word. You are trying to say that "we support additional features that add to the server functionality" yes?

Reply:
Yes, no need to call it middleware here. Callout was mainly around how we have a pattern for creating server-side features

Reply:
Ok, let's workshop the name so we don't give the wrong impression.

Reply:
Can we strip 

> Internal middleware supported to augment functionality

For now?
```suggestion
| Server             | Returns HTML templates for routes that bootstrap the application (features like SSR possible in future). Configurable via CLI args/flags or environment variables.                                                 | Returns HTML page with client bundle and preloaded route federation `remoteEntry`.    |
```